packages/store/src/internal/state-context-factory.ts

```diff
@@ -27,6 +27,18 @@ export class StateContextFactory {
 
     function setStateValue(currentAppState: any, newValue: T): any {
       const newAppState = setValue(currentAppState, metadata.depth, newValue);
+
+      if (metadata.instance && metadata.instance.ngxsOnChanges) {
+        const instance: NgxsLifeCycle = metadata.instance;
+        instance.isFirstChange = true;
```

Comment:
i find this a bit confusing, since we are setting it to true, when its actually gonna be false

Reply:
it should always be true after ngxsOnInit

Reply:
done